### PR TITLE
libkbfs: actually keep a favorites cache

### PR DESCRIPTION
**Pull request by jakob223**
This commit changes the favorites cache to not refresh until it is invalidated instead of refreshing every time.

Relevant PRs:
 - https://github.com/keybase/client/pull/14370
 - https://github.com/keybase/keybase/pull/3059

### CODE REVIEW

File: libkbfs/favorites.go [hunk 1]
```diff
@@ -340,8 +338,9 @@ func (f *Favorites) RefreshCache(ctx context.Context) {
 	}
 }
 
-// Get returns the logged-in users list of favorites. It
-// doesn't use the cache.
+// Get returns the logged-in user's list of favorites. It uses the cache.
+// TODO: decide on behavior if we've gotten an invalidation but not an updated
```
Comment (strib) [inline libkbfs/favorites.go]: What are the choices here?  If we've gotten an invalidation, I think the `Get` should block on fetching the favorites (which is what should be happening in this PR if I'm reading the code right.)

**Comment by jakob223** [comment]
That's what we're currently doing - I was just concerned that we don't want
to block on fetching the favorites if we're offline - there's a situation
whereby we get invalidated and then get offline before the new favorites
arrive and then can't get the favorites at all. But it's a pretty unlikely
event.

**Comment by jakob223** [comment]
Oops, when I said "forthcoming" I really meant "there's another PR for that that I've now updated.

The newest commit invalidates the favorites cache after a week, for reasons described here: https://github.com/keybase/client/pull/14554

File: libkbfs/favorites.go [hunk 2]
```diff
@@ -120,15 +123,16 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
...
-	if req.refresh || f.cache == nil || req.favs != nil {
+	//  * It's stale
+	if req.refresh || f.cache == nil || time.Now().After(f.cacheExpireTime) {
 		folders, err := kbpki.FavoriteList(req.ctx)
 		if err != nil {
 			return err
 		}
 
 		f.cache = make(map[Favorite]bool)
+		f.cacheExpireTime = libkb.ForceWallClock(time.Now()).Add(time.
```
Comment (strib) [inline libkbfs/favorites.go]: Please make the expire time interval into a global const in this file, for easier tweaking in the future.

File: libkbfs/favorites.go (hunk 1 above)
Comment (strib) [reply libkbfs/favorites.go]: > That's what we're currently doing - I was just concerned that we don't want to block on fetching the favorites if we're offline - there's a situation whereby we get invalidated and then get offline before the new favorites arrive and then can't get the favorites at all. But it's a pretty unlikely event.

In this case, the `FavoriteList` call will just fail, and the cache will remain in place.  Maybe in that case, when `req.refresh == true`, you can just set the expiration time to a few minutes in the future?  (And then get rid of this TODO.)

**Pull request merged**